packages/compose/src/lazy-load/index.js

```diff
@@ -0,0 +1,78 @@
+import { Component } from '@wordpress/element';
+
+const lazyLoad = ( scripts, styles, baseUrl ) => ( Wrapped ) => {
+	function loadScript() {
+		return new Promise( ( resolve, reject ) => {
+			const script = document.createElement( 'script' );
+			script.src = baseUrl + '/wp-admin/load-scripts.php?load=' + scripts.join( ',' );
+			script.onload = resolve;
+			script.onerror = reject;
+
+			document.head.appendChild( script );
+		} );
+	}
+
+	function loadStyle() {
+		return new Promise( ( resolve, reject ) => {
+			const style = document.createElement( 'link' );
+			style.rel = 'stylesheet';
+			style.href = baseUrl + '/wp-admin/load-styles.php?load=' + styles.join( ',' );
+			style.onload = resolve;
+			style.onerror = reject;
+
+			document.head.appendChild( style );
+		} );
+	}
+
+	let hasAlreadyLoadedAssets = false;
+
+	function loadAssets() {
+		if ( hasAlreadyLoadedAssets ) {
```

Comment:
This won't handle the case where another instance of the component is mounted _while_ loading is still occurring.

We should just set the scope variable to the promise itself. That way it can repeatedly be `.then`'d, even if it's already resolved.

Reply:
Should there be retry if a previous load attempt failed?